docs/api/index.md

```diff
@@ -512,16 +511,205 @@ TODO
+
+  const invoice = {
+    isActive: true,
+    customer: {
+      first_name: 'John',
+      last_name: 'Doe',
+      location: 'China',
+    },
+    total_amount: 5000,
+    items: [
+      {
+      type: 'apples',
+      quantity: 10,
+      },
+      {
+      type: 'oranges',
+      quantity: 5,
+      },
+    ]
+  }
+
+  test('John Doe Invoice', () => {
+    expect(invoice).toHaveProperty('isActive') // assert that the key exists
+    expect(invoice).toHaveProperty('total_amount', 5000) //assert that the key exists and the value is equal
+
+    expect(invoice).not.toHaveProperty('account') //assert that this key does not exist
+
+    // Deep referencing using dot notation
+    expect(invoice).toHaveProperty('customer.first_name')
+    expect(invoice).toHaveProperty('customer.last_name', 'Doe')
+    expect(invoice).not.toHaveProperty('customer.location', 'India')
+
+    // Deep referencing using an array containing the key
+    expect(invoice).toHaveProperty('items[0].type', 'apples')
+  })
+  ```
+
 ### toMatch
+
+- **Type:** `(received: string | regexp) => Awaitable<void>`
+
+  `toMatch` asserts if a string matches a regular expression or a string.
+
+  ```ts
+  import { expect, test } from 'vitest'
+
+  test('top fruits', () => {
+    expect('top fruits include apple, orange and grape').toMatch(/apple/)
+    expect('applefruits').toMatch('fruit') // toMatch also accepts a string
+  })
+  ```
+
 ### toMatchObject
 
+- **Type:** `(received: object | array) => Awaitable<void>`
+
+  `toMatchObject` asserts if an object matches a subset of the properties of an object.
+
+  You can also pass an array of objects. This is useful if you want to check that two arrays match in their number of elements, as opposed to `arrayContaining`, which allows for extra elements in the received array.
+
+  ```ts
+  import { test, expect } from 'vitest'
+
+  const johnInvoice = {
+    isActive: true,
+    customer: {
+      first_name: 'John',
+      last_name: 'Doe',
+      location: 'China',
+    },
+    total_amount: 5000,
+    items: [
+      {
+      type: 'apples',
+      quantity: 10,
+      },
+      {
+      type: 'oranges',
+      quantity: 5,
+      }
+    ]
+  }
+
+  const johnDetails = {
+    customer: {
+      first_name: 'John',
+      last_name: 'Doe',
+      location: 'China',
+    }
+  }
+
+  test('invoice has john personal details', () => {
+    expect(johnInvoice).toMatchObject(johnDetails)
+  })
+
+  test('the number of elements must match exactly', () => {
+    // Assert that an array of object matches
+    expect([{ foo: 'bar' }, { baz: 1 }]).toMatchObject([
+      { foo: 'bar' },
+      { baz: 1 },
+    ])
+  })
+  ```
+
 ### toThrowError
 
+- **Type:** `(received: any) => Awaitable<void>`
+
+  `toThrowError` asserts if a function throws an error when it is called.
+
+  For example, if we want to test that getFruitStock('pineapples') throws, because pineapples is not good for people with diabetes, we could write:
```

Comment:
Should be before the example:
```suggestion
  For example, if we want to test that `getFruitStock('pineapples')` throws, because pineapples is not good for people with diabetes, we could write:
```